From: Thorsten Behrens <tbehrens@suse.com>
Date: Mon, 1 Dec 2014 13:14:54 +0100
Subject: [PATCH] Fix spec-violating extraneous unlock in RWLock dtor.

The test/cli/radosgw-admin/help.t crashed on a number of newer glibc
versions below RGWCache<RGWRados>::~RGWCache(), due to
pthread_rw_unlock being called w/o prior lock. POSIX says this is
no-no.

Signed-off-by: Thorsten Behrens <tbehrens@suse.com>
(cherry picked from commit 8ad97911b681e881b996b11c283741241c7e782d)
---
 src/common/RWLock.h | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/common/RWLock.h b/src/common/RWLock.h
index f901ac0..b7621c0 100644
--- a/src/common/RWLock.h
+++ b/src/common/RWLock.h
@@ -36,7 +36,6 @@ public:
   }
 
   virtual ~RWLock() {
-    pthread_rwlock_unlock(&L);
     pthread_rwlock_destroy(&L);
   }
 
